### PR TITLE
MISUV-7776 Remove old style Authentication for agents

### PR DESCRIPTION
**Pull request by danielmaltz1**

### CODE REVIEW

File: app/controllers/FinalTaxCalculationController.scala [hunk 1]
```diff
@@ -108,17 +99,22 @@ class FinalTaxCalculationController @Inject()(implicit val cc: MessagesControlle
...
   }
 
-  def agentSubmit(taxYear: Int): Action[AnyContent] = Authenticated.async { implicit request =>
-    implicit agent =>
-      getMtdItUserWithIncomeSources(incomeSourceDetailsService).flatMap { user =>
-        val fullName = user.session.get(clientFirstName).getOrElse("") + " " + user.session.get(clientLastName).getOrElse("")
-        agentFinalDeclarationSubmit(taxYear, fullName)(user, hc)
-      }
-  }
+//  def agentSubmit(taxYear: Int): Action[AnyContent] = auth.authenticatedAction(isAgent = true) { user =>
```
Comment (danielmaltz1) [inline app/controllers/FinalTaxCalculationController.scala:102]: Using auth.authenticatedAction here causes a unit test to fail in FinalTaxCalculationControllerSpec

File: app/controllers/agent/ConfirmClientUTRController.scala [hunk 2]
```diff
@@ -33,15 +34,25 @@ import scala.concurrent.{ExecutionContext, Future}
...
+//    implicit user =>
+//      Future.successful(Ok(confirmClient(
+//        clientName = fetchClientName,
+//        clientUtr = fetchClientUTR,
+//        postAction = routes.ConfirmClientUTRController.submit,
+//        backUrl = backUrl
+//      )))
+//  }
+
+  def show: Action[AnyContent] = auth.authenticatedAction(isAgent = false) { implicit user =>
```
Comment (danielmaltz1) [inline app/controllers/agent/ConfirmClientUTRController.scala:55]: Using auth.authenticatedAction here causes an AT to fail

File: app/controllers/CalculationPollingController.scala [hunk 3]
```diff
@@ -28,21 +28,22 @@ import play.api.mvc._
 import services.CalculationPollingService
 import uk.gov.hmrc.auth.core.AuthorisedFunctions
 import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.language.LanguageUtils
```
Comment (avelx) [inline app/controllers/CalculationPollingController.scala:31]: pls optimise imports

File: app/controllers/ChargeSummaryController.scala [hunk 4]
```diff
@@ -48,20 +48,15 @@ object ChargeSummaryController {
   case class ErrorCode(message: String)
 }
 
-class ChargeSummaryController @Inject()(val authenticate: AuthenticationPredicate,
-                                        val auth: AuthenticatorPredicate,
-                                        val checkSessionTimeout: SessionTimeoutPredicate,
-                                        val retrieveNinoWithIncomeSources: IncomeSourceDetailsPredicate,
+class ChargeSummaryController @Inject()(val auth: AuthenticatorPredicate,
```
Comment (avelx) [inline app/controllers/ChargeSummaryController.scala:51]: Please optimise imports in this file

File: app/controllers/ForecastIncomeSummaryController.scala [hunk 5]
```diff
@@ -34,41 +34,36 @@ import services.{CalculationService, IncomeSourceDetailsService}
 import uk.gov.hmrc.auth.core.AuthorisedFunctions
 import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.language.LanguageUtils
+import utils.AuthenticatorPredicate
```
Comment (avelx) [inline app/controllers/ForecastIncomeSummaryController.scala:37]: Please optimise imports in this file

File: app/controllers/PaymentController.scala [hunk 6]
```diff
@@ -28,27 +28,25 @@ import play.api.Logger
 import play.api.mvc._
 import uk.gov.hmrc.auth.core.AffinityGroup
 import uk.gov.hmrc.auth.core.AffinityGroup.Agent
+import utils.AuthenticatorPredicate
```
Comment (avelx) [inline app/controllers/PaymentController.scala:31]: Optimise imports pls

File: app/utils/AuthenticatorPredicate.scala [hunk 7]
```diff
@@ -16,16 +16,17 @@
...
 
-import auth.MtdItUser
+import auth.{MtdItUser, MtdItUserBase, MtdItUserOptionNino, MtdItUserWithNino}
 import config.featureswitch.FeatureSwitching
 import config.{AgentItvcErrorHandler, FrontendAppConfig}
 import controllers.agent.predicates.ClientConfirmedController
 import controllers.predicates._
-import play.api.i18n.I18nSupport
-import play.api.mvc.{Action, AnyContent, MessagesControllerComponents, Result}
+import play.api.i18n.{I18nSupport, Messages}
```
Comment (avelx) [inline app/utils/AuthenticatorPredicate.scala:24]: optimise imports

File: app/utils/AuthenticatorPredicate.scala [hunk 8]
```diff
@@ -66,4 +69,27 @@ class AuthenticatorPredicate @Inject()(val checkSessionTimeout: SessionTimeoutPr
         authenticatedCodeBlock(user)
       }
   }
+
```
Comment (avelx) [inline app/utils/AuthenticatorPredicate.scala:72]: Please apply code formatting in this file

**Review by avelx — APPROVED** [review]
approved with comments

**Review by avelx — APPROVED** [review]

**Pull request merged**